### PR TITLE
OTT-168 : Use VAST unwrap traffic percentage from UI if VAST Unwrap is enabled

### PR DESCRIPTION
**Pull request by Ankit-Pinge**

### CODE REVIEW

File: modules/pubmatic/openwrap/openwrap_sshb.go [hunk 1]
```diff
@@ -54,8 +54,10 @@ func GetVastUnwrapEnabled(rctx vastmodels.RequestCtx) bool {
 	}
 	partnerConfigMap, err := ow.getProfileData(rCtx, openrtb2.BidRequest{})
 	if err != nil || len(partnerConfigMap) == 0 {
-		return false
+		return false, ""
 	}
 	rCtx.PartnerConfigMap = partnerConfigMap
-	return models.GetVersionLevelPropertyFromPartnerConfig(rCtx.PartnerConfigMap, models.VastUnwrapperEnableKey) == VastUnwrapperEnableValue
+	unwrapEnabled := models.GetVersionLevelPropertyFromPartnerConfig(rCtx.PartnerConfigMap, models.VastUnwrapperEnableKey)
+	trafficPercent := models.GetVersionLevelPropertyFromPartnerConfig(rCtx.PartnerConfigMap, models.VastUnwrapTrafficPercentKey)
```
Comment (pm-jaydeep-mohite) [inline modules/pubmatic/openwrap/openwrap_sshb.go:61]: Can we take decision here only and return bool 

```
if unwrapEnabled == "1"  {
              trafficPercent := models.GetVersionLevelPropertyFromPartnerConfig(rCtx.PartnerConfigMap, models.VastUnwrapTrafficPercentKey)
              trafficPercentageInt, err := strconv.Atoi(trafficPercentage)
 		if err == nil {
 			vastUnwrapEnabled = getRandomNumber() < trafficPercentageInt && unwrapEnabled
 			return vastUnwrapEnabled
 		}
}
return false 
```

File: modules/pubmatic/openwrap/openwrap_sshb.go [hunk 2]
```diff
@@ -54,8 +54,10 @@ func GetVastUnwrapEnabled(rctx vastmodels.RequestCtx) bool {
 	}
 	partnerConfigMap, err := ow.getProfileData(rCtx, openrtb2.BidRequest{})
 	if err != nil || len(partnerConfigMap) == 0 {
-		return false
+		return false, ""
 	}
 	rCtx.PartnerConfigMap = partnerConfigMap
-	return models.GetVersionLevelPropertyFromPartnerConfig(rCtx.PartnerConfigMap, models.VastUnwrapperEnableKey) == VastUnwrapperEnableValue
+	unwrapEnabled := models.GetVersionLevelPropertyFromPartnerConfig(rCtx.PartnerConfigMap, models.VastUnwrapperEnableKey)
```
Comment (AvinashKapre) [inline modules/pubmatic/openwrap/openwrap_sshb.go:60]: Please add the UT's for this.

File: modules/pubmatic/openwrap/openwrap_sshb.go (hunk 1 above)
Comment (pm-jaydeep-mohite) [reply modules/pubmatic/openwrap/openwrap_sshb.go:61]: And update _getRandomNumber_ as which will return numbers 1 to 100 

```
var getRandomNumber = func() int {
	return rand.Intn(100) + 1
} 
```

File: modules/pubmatic/openwrap/openwrap_sshb.go [hunk 3]
```diff
@@ -44,7 +46,7 @@ func (ow *OpenWrap) SetMetricEngine(m metrics.MetricsEngine) {
 	ow.metricEngine = m
 }
 
-// GetVastUnwrapEnabled function return vastunwrap flag from the database
+// GetVastUnwrapEnabled returns whether to enable vastunwrap o
```
Comment (pm-jaydeep-mohite) [inline modules/pubmatic/openwrap/openwrap_sshb.go:49]: Update comment

File: modules/pubmatic/openwrap/openwrap_sshb.go (hunk 2 above)
Comment (Ankit-Pinge) [reply modules/pubmatic/openwrap/openwrap_sshb.go:62]: Added UT's

File: modules/pubmatic/openwrap/openwrap_sshb.go (hunk 3 above)
Comment (Ankit-Pinge) [reply modules/pubmatic/openwrap/openwrap_sshb.go]: updated

**Review by AvinashKapre — APPROVED** [review]

**Review by pm-jaydeep-mohite — APPROVED** [review]
LGTM

**Pull request merged**